src/batch.ts

```diff
@@ -0,0 +1,74 @@
+import { ethers } from 'ethers';
+import {
+  checkAddress,
+  getMetropolisContract,
+  handleEthersError,
+  getIsSameVersion,
+} from './lib/utils';
+import Pod from './Pod';
+
+/**
+ * Batch transfers pod memberships. If a signer is provided, it will execute the transaction. Otherwise it will return the unsigned tx.
+ *
+ * @param fromAddress - Address that is giving up membership
+ * @param toAddress - Address that will receive membership
+ * @param podIds - Array of pod IDs for which the toAddress will receive membership
+ * @param signer - If a signer is provided, then the tx will execute. Otherwise, an unsigned transaction will be returned.
+ * @throws If toAddress is already a member
+ * @throws if fromAddress is not a member
+ * @throws If provided signer is not the fromAddress
+ */
+export default async function batchTransferMembership(
+  fromAddress: string,
+  toAddress: string,
+  podIds: number[],
+  signer?: ethers.Signer,
+): Promise<ethers.providers.TransactionResponse | { to: string; data: string }> {
+  // we need to get the pod object for the podIds for the member and version checks
+  const pods = await Promise.all(podIds.map(async podId => new Pod(podId)));
+  // check the fromAddress
+  const checkedFrom = checkAddress(fromAddress);
+  // check the toAddress
+  const checkedTo = checkAddress(toAddress);
```

Comment:
we also need to add a check to make sure the `toAddress` isn't already a member of all the pods